app/src/main/java/ru/flicksbox/network/NetworkManager.kt

```diff
@@ -0,0 +1,48 @@
+package ru.flicksbox.network
+
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import okhttp3.OkHttpClient
+import okhttp3.ResponseBody
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Converter
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import kotlin.math.log
+
+interface NetworkManager {
+    fun <T> createManager(clazz: Class<T>): T
+}
+
+class NetworkManagerImpl : NetworkManager {
+    companion object {
+        const val API_ENDPOINT = "https://www.flicksbox.ru"
+    }
+
+    private val okHttpClient: OkHttpClient by lazy {
+        OkHttpClient.Builder().addInterceptor(loggingInterceptor).build()
+    }
+
+    private val gson: Gson by lazy {
+        GsonBuilder().create()
+    }
+
+    private val converterFactory: Converter.Factory by lazy {
+        GsonConverterFactory.create(gson)
+    }
+
+    private val loggingInterceptor: HttpLoggingInterceptor by lazy {
+        HttpLoggingInterceptor().apply {
+            level = HttpLoggingInterceptor.Level.BODY
+        }
+    }
+
+    val retrofit: Retrofit by lazy {
+        Retrofit.Builder().addConverterFactory(converterFactory).baseUrl(API_ENDPOINT)
+            .client(okHttpClient).build()
+    }
+
+    override fun <T> createManager(clazz: Class<T>): T =
```

Comment:
you actually creating Services, not Managers
I suggest to rename the method to `createService` and `NetworkManager` to `NetworkServiceFactory` or sth like that
It's absolutely not about managers